### PR TITLE
Added verification for server config

### LINKED ISSUE #17: validate uploaded server configuration to ensure proper syntax

### LINKED ISSUE #83: [Bug]: Using /new-user-purchase does not encrypt the password
### Describe the bug
When using `/new-user-purchase` the new user password is not encrypted

### Expected behavior
Create password to be encrypted

### Actual behavior
User password is store in plain text

### LINKED ISSUE #84: [Bug]: Incorrect paths in Agent
### Describe the bug
Path configure in `agent` are not all the same when running command or when accessing resources/files.
All current path are relative and not absolute which are not that reliable 

### Expected behavior

Paths to route to the correct resources

### Actual behavior

Current path are now guarantee to be

### PR DESCRIPTION
**Pull request by Arturo-Lopez**
# Related Issue/Addition to code

- Fixes #17, Fixes #83, Fixes #84

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# Proposed Changes
- Added yup to validate server configuration changes
- Added a blacklist of values to be not allowed to be change `['server-port', 'server-ip]';`
- Modified client server configuration form in order to replace string inputs when needed
- Fixed multiple bad paths in agent
- Replace node-fetch due to ESM being incompatible. [Link problem](https://github.com/node-fetch/node-fetch/issues/1279)
- Fixed user password not being hashed
- Remove unused files

### CODE REVIEW

File: agent/src/commands/runBackup.ts [hunk 1]
```diff
@@ -20,13 +20,13 @@ export const runBackup: runBackupInterface = async ({ serverId }) => {
   await exec(`screen -S "${serverId}" -p 0 -X stuff "/save-all\r"`);
 
   // create the tar
-  await exec(`cd ../../servers/${serverId} && tar -zcvf ${serverId}.tar.gz .`);
+  await exec(`cd $SERVERS_DIR/${serverId} && tar -zcvf ${serverId}.tar.gz .`);
```
Comment (codyseibert) [inline agent/src/commands/runBackup.ts:23]: Where is this defined?  Shouldn't this be a process.env variable somewhere?

File: agent/src/commands/runBackup.ts [hunk 2]
```diff
@@ -20,13 +20,13 @@ export const runBackup: runBackupInterface = async ({ serverId }) => {
...
 
   // TODO: upload  to s3 instead of copying to tmp
-  await exec(`cp ../../servers/${serverId}/${serverId}.tar.gz ..`);
+  await exec(`cp $SERVERS_DIR/${serverId}/${serverId}.tar.gz ..`);
```
Comment (codyseibert) [inline agent/src/commands/runBackup.ts:26]: here as well

File: agent/src/commands/runBackup.ts [hunk 3]
```diff
@@ -20,13 +20,13 @@ export const runBackup: runBackupInterface = async ({ serverId }) => {
...
 
   // delete the tar
-  await exec(`rm ../../servers/${serverId}/${serverId}.tar.gz`);
+  await exec(`rm $SERVERS_DIR/${serverId}/${serverId}.tar.gz`);
```
Comment (codyseibert) [inline agent/src/commands/runBackup.ts:29]: here as well

File: agent/src/commands/stopOrphanedServers.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ export const stopOrphanedServers: stopOrphanedServersInterface = async ({
       // eslint-disable-next-line no-await-in-loop
       promises.push(
         exec(`docker stop --time=30 ${runningId}`).then(() =>
-          exec(`rm -rf ../../servers/${runningId}`),
+          exec(`rm -rf $SERVERS_DIR/${runningId}`),
```
Comment (codyseibert) [inline agent/src/commands/stopOrphanedServers.ts:28]: here as well

**Review by codyseibert — CHANGES_REQUESTED** [review]

File: agent/src/commands/runBackup.ts (hunk 1 above)
Comment (Arturo-Lopez) [reply agent/src/commands/runBackup.ts:23]: This is works the same as the process.env. The SERVER_DIR is added to the system environment at the start time of the Node Js process and can be accessed via code with the process.env or with $SERVERS_DIR in the context of executing command directly with exec. The alternative of this is putting `path.join(process.env.SERVERS_DIR, "/${serverId}")`

**Pull request merged**